### ISSUE TITLE
The resently added EnumerableResolutionStrategy does not seem to work.

### ISSUE THREAD

**Issue by srgoapp** [NancyFx/Nancy.Bootstrappers.Unity#11]
The projects in the solution won't build. Several files are missing.

Appart from that, i think there is a problem with the recent change about EnumerableResolutionStrategy.

Try the sample on this site:

http://www.stefanbader.ch/minimal-sample-self-hosted-nancy-using-owin-unity-bootstrapper-including-xunit-test/

and update the NuGet-package to 1.2. It will throw the following exception:

Resolution of the dependency failed, type = "Nancy.Bootstrapper.IApplicationStartup", name = "Nancy.ViewEngines.ViewEngineApplicationStartup". Exception occurred while: while resolving.

Exception is: InvalidOperationException - The current type, System.Collections.Generic.IEnumerable`1[Nancy.ViewEngines.IViewEngine], is an interface and cannot be constructed. Are you missing a type mapping?

At the time of the exception, the container was:

Resolving Nancy.ViewEngines.ViewEngineApplicationStartup,Nancy.ViewEngines.ViewEngineApplicationStartup (mapped from Nancy.Bootstrapper.IApplicationStartup, Nancy.ViewEngines.ViewEngineApplicationStartup) Resolving parameter "viewEngines" of constructor Nancy.ViewEngines.ViewEngineApplicationStartup(System.Collections.Generic.IEnumerable1[[Nancy.ViewEngines.IViewEngine, Nancy, Version=1.2.0.0, Culture=neutral, PublicKeyToken=null]] viewEngines, Nancy.ViewEngines.IViewCache viewCache, Nancy.ViewEngines.IViewLocator viewLocator)
    Resolving System.Collections.Generic.IEnumerable1[Nancy.ViewEngines.IViewEngine],(none)

The current type, System.Collections.Generic.IEnumerable`1[Nancy.ViewEngines.IViewEngine], is an interface and cannot be constructed. Are you missing a type mapping?

**Comment by thecodejunkie** [comment]
@khellang is this related to the PR you sent a while back?

**Comment by tseel** [comment]
I experienced the same issue. Is there any workaround or fix in sight?

**Comment by baddonkey** [comment]
Taking the sample from http://www.stefanbader.ch/minimal-sample-self-hosted-nancy-using-owin-unity-bootstrapper-including-xunit-test , for Version 1.2. you need to slightly adapt the customized bootstrapper:
```csharp
using Microsoft.Practices.Unity;
using Nancy.Bootstrappers.Unity;

namespace NancyOwinBox
{
    public class NancyOwinBoxBootstrapper : UnityNancyBootstrapper
    {
        private IUnityContainer _container;

        public NancyOwinBoxBootstrapper(IUnityContainer container)
        {
            _container = container;
            _container.AddNewExtension<EnumerableExtension>();
        }

        protected override IUnityContainer GetApplicationContainer()
        {
            return _container;
        }
    }
}
```
Line:
```csharp
_container.AddNewExtension<EnumerableExtension>();
```
is doing the trick.

**Comment by khellang** [comment]
@thecodejunkie Yes, I think this might be related. As @baddonkey mentioned, it should work when calling `AddNewExtension<EnumerableExtension>`, but we already do that [here](https://github.com/NancyFx/Nancy.Bootstrappers.Unity/blob/69cf899034ac13bf3a77f13898b78550b4f11e85/src/Nancy.Bootstrappers.Unity/UnityNancyBootstrapper.cs#L86), so it must be something else. Could it be that we have to do the same call for the child (request) container? @tseel @srgoapp Does it work if you call `AddNewExtension<EnumerableExtension>` on the request container in `ConfigureRequestContainer`?

**Comment by khellang** [comment]
Ah, now I see what's happening here. You're using a container that's not constructed by Nancy. Then you have to manually add the extension, like @baddonkey says.

**Comment by khellang** [comment]
I have a pending fix for this at #12. This'll make it "just work" without having to manually add the extension, even if you're using an existing, "external" bootstrapper.

**Comment by khellang** [comment]
The fix in #12 will go out with the 1.3 release. Hopefully next week :+1:
